lib/slop/result.rb

```diff
@@ -33,16 +43,17 @@ def []=(flag, value)
 
     # Returns an Option if it exists. Ignores any prefixed hyphens.
     def option(flag)
-      cleaned = -> (f) do
-        key = f.to_s.sub(/\A--?/, '')
-        key = key.tr '-', '_' if parser.config[:underscore_flags]
-        key.to_sym
-      end
       options.find do |o|
-        o.flags.any? { |f| cleaned.(f) == cleaned.(flag) }
+        o.flags.any? { |f| clean_key(f) == clean_key(flag) }
       end
     end
 
+    def clean_key(key)
```

Comment:
If you decide to keep this method, please move it to the bottom of the class and mark it as `private`. It should only be used internally

Reply:
Done!